drivers: mipi_dbi: mipi_dbi_spi: change reset pin polarity

danieldegrasse opened the pull request:
Change reset pin polarity for MIPI DBI SPI controller, so that the board
devicetree is responsible for setting the GPIO to active low, and the
driver always sets the pin to a logic 1 to reset the display.

Fixes #68562

danieldegrasse committed:
drivers: mipi_dbi: mipi_dbi_spi: change reset pin polarity

Change reset pin polarity for MIPI DBI SPI controller, so that the board
devicetree is responsible for setting the GPIO to active low, and the
driver always sets the pin to a logic 1 to reset the display.

Fixes #68562

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

fabiobaltieri reviewed (dismissed):
Thanks!

diegoherranz commented:
Thanks for updating this.

This line needs updating too: https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/mipi-dbi/zephyr%2Cmipi-dbi-spi.yaml#L29

fabiobaltieri commented:
Tagging for 3.6 since `drivers/mipi_dbi/mipi_dbi_spi.c` was introduced in this cycle, saves a breaking change against the next release.

danieldegrasse commented:
> This line needs updating too: https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/mipi-dbi/zephyr%2Cmipi-dbi-spi.yaml#L29. Simply remove "Active low pin."

Description has been updated to clarify pin will be driven high to reset the display

MaureenHelm commented on dts/bindings/mipi-dbi/zephyr,mipi-dbi-spi.yaml:
```diff
@@ -25,8 +25,7 @@ properties:
     description: |
-      Reset GPIO pin. Used to reset the display during initialization.
-      Active low pin.
+      Reset GPIO pin. Set high to reset the display
```
This change should be in the previous commit, not the migration guide commit

danieldegrasse replied:
Done, thanks for catching this

fabiobaltieri commented on boards/arm/bl5340_dvk/bl5340_dvk_cpuapp_common.dtsi:
```diff
@@ -110,7 +110,7 @@
 		compatible = "zephyr,mipi-dbi-spi";
-		reset-gpios = <&gpio0 6 GPIO_ACTIVE_HIGH>;
+		reset-gpios = <&gpio0 6 GPIO_ACTIVE_LOW>;
 		dc-gpios = <&gpio0 12 GPIO_ACTIVE_HIGH>;
```
hey @danieldegrasse, sorry for adding to this PR, I just converted one of my board and noticed that dc also changed polarity from the previous convention -- for this one I think it's fine to use it as active high, but if you want to keep it like that it's a good idea to update the migration guide example (https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/releases/migration-guide-3.6.rst?plain=1#L271-L272) and maybe even mention it separately

danieldegrasse replied:
No worries! I noticed that as well, and I made the change in this PR. I can split it into another PR if you think that would be best though. I'll update the commit in this PR to call out the fact that the DC pin changed polarity

fabiobaltieri replied:
nah cool to have it here, you have a duplicate Signed-off-by in the commit message though in case you care about it

danieldegrasse replied:
I do, thanks for noticing- I'll fix it

danieldegrasse committed:
doc: migration-guide-3.6: update MIPI DBI display migration note

Update MIPI DBI display migration note to match the correct display
configuration settings for reset and MIPI DBI command pins after ILI9xxx
display was transitioned to this API.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>
